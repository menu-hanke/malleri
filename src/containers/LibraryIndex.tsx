import React, {useState} from 'react';
import {ModelFunctionPackage} from '../../types';

export interface Props {
  packages: ModelFunctionPackage[]
}

export default (props: Props) => {
  // 1. search input and filtering
  // 2. refactoring & unit testing functionality
  // 3. new view for model function package

  const [titleFilterInput, setTitleFilterInput] = useState('');
  const [descriptionFilterInput, setDescriptionFilterInput] = useState('');

  const totalFilter = (item: ModelFunctionPackage) => {
    return item.title.includes(titleFilterInput) && item.description.includes(descriptionFilterInput);
  };

  const applyPackagesFiltering = (packages: ModelFunctionPackage[]) => {
    return packages.filter(totalFilter);
  };

  return (
    <div>
      <input type="text" placeholder="Filter by title..." onChange={(event) => setTitleFilterInput(event.target.value)} />
      <input type="text" placeholder="Filter by description..." onChange={(event) => setDescriptionFilterInput(event.target.value)} />
      <ul>
        {applyPackagesFiltering(props.packages).map((item: ModelFunctionPackage, i: number) => (
          <li key={i}>{item.title}</li>
        ))}
      </ul>
    </div>
  );
};
