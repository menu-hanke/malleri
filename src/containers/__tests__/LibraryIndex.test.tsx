import React from 'react';
import renderer from 'react-test-renderer';
import data from '../../../public/mfk-presentation-level-collection.json';

import LibraryIndex, {Props} from '../LibraryIndex';

const renderTree = () => {
  const defaultProps: Props = {
    packages: data
  };

  return renderer.create(<LibraryIndex {...defaultProps} />);
};


test('snapshot should match', async () => {
  const resultElement = renderTree();
  expect(resultElement.toJSON()).toMatchSnapshot();
});
