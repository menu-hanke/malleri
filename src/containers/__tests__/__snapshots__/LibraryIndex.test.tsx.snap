// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Filter by title..."
    type="text"
  />
  <table
    className="sc-bdnxRM hdzVvY"
  >
    <tbody>
      <tr>
        <td
          className="sc-gtsrHT gBukoy"
        >
          <table>
            <tr
              onClick={[Function]}
            >
              <td
                style={
                  Object {
                    "width": "20px",
                  }
                }
              >
                <b>
                  +
                </b>
              </td>
              <td>
                Natural processes
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td
          className="sc-gtsrHT gBukoy"
        >
          <table>
            <tr
              onClick={[Function]}
            >
              <td
                style={
                  Object {
                    "width": "20px",
                  }
                }
              >
                <b>
                  +
                </b>
              </td>
              <td>
                Tree models
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td
          className="sc-gtsrHT gBukoy"
        >
          <table>
            <tr
              onClick={[Function]}
            >
              <td
                style={
                  Object {
                    "width": "20px",
                  }
                }
              >
                <b>
                  +
                </b>
              </td>
              <td>
                Tree volume calculations
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td
          className="sc-gtsrHT gBukoy"
        >
          <table>
            <tr
              onClick={[Function]}
            >
              <td
                style={
                  Object {
                    "width": "20px",
                  }
                }
              >
                <b>
                  +
                </b>
              </td>
              <td>
                Initial data conversion
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td
          className="sc-gtsrHT gBukoy"
        >
          <table>
            <tr
              onClick={[Function]}
            >
              <td
                style={
                  Object {
                    "width": "20px",
                  }
                }
              >
                <b>
                  +
                </b>
              </td>
              <td>
                Berry production
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
